org.eclipse.xtext/src/org/eclipse/xtext/validation/NamesAreUniqueValidator.java

```diff
@@ -62,18 +69,18 @@ public void checkUniqueNamesInResourceOf(EObject eObject) {
 	}
 
 	public void doCheckUniqueNames(Resource resource, CancelIndicator cancelIndicator) {
-		final IResourceServiceProvider resourceServiceProvider = resourceServiceProviderRegistry.getResourceServiceProvider(resource.getURI());
-		if (resourceServiceProvider==null)
-			return;
-		IResourceDescription.Manager manager = resourceServiceProvider.getResourceDescriptionManager();
-		if (manager != null) {
-			IResourceDescription description = manager.getResourceDescription(resource);
-			if (description != null) {
-				Iterable<IEObjectDescription> descriptions = description.getExportedObjects();
-				helper.checkUniqueNames(descriptions, cancelIndicator, this);
-			}
+		Context validationContext = getValidationContext(resource, cancelIndicator);
+		if (validationContext != null) {
+			helper.checkUniqueNames(validationContext, this);
 		}
 	}
+	
+	/**
+	 * @since 2.22
+	 */
+	protected INamesAreUniqueValidationHelper.Context getValidationContext(Resource resource, CancelIndicator cancelIndicator) {
+		return contextProvider.tryGetContext(resource, cancelIndicator);
```

Comment:
null check?

Reply:
Made it more resilient against old configurations